plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -310,6 +318,20 @@ protected String getTableSchemaName(ResultSet resultSet)
 
             case Types.DATE:
                 return Optional.of(dateColumnMapping());
+
+            case Types.TIMESTAMP:
+                // support higher precisions (TODO https://github.com/trinodb/trino/issues/6910)
+                TimestampType timestampType = TIMESTAMP_MILLIS;
+                return Optional.of(ColumnMapping.longMapping(
+                        timestampType,
+                        (resultSet, columnIndex) -> {
+                            Timestamp timestamp = resultSet.getTimestamp(columnIndex);
```

Comment:
BTW, if the driver is JDBC 4.2 compatible, then you should be able use use:
```
Instant instant = resultSet.getObject(columnIndex, Instant.class);
```

Reply:
We'd need to upgrade the MySQL driver version - which I tried but I need to add some more tests to make sure we don't break anything (notably MariaDB) in the process.

Reply:
Are we really using that old a version?

Anyway, it looks like MySQL's `getObject(i, LocalDateTime.class)` just calls `getTimestamp(i).toLocalDateTime()`, so that should work just as well.

Reply:
The version we are on does't have handling for any `java.time` parameters to `getObject`.

Also, a notable bug https://bugs.mysql.com/bug.php?id=93444 (so I'll need to bump to a version containing this fix once I do bump the version).